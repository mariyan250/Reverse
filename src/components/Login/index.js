// React and Style
import React, { useState } from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Components
import AuthInput from '../AuthInput';
import AuthButton from '../AuthButton';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = (e) => {
    e.preventDefault();
    console.log({ email, password });
  }

  return (
    <main className="login d-md-flex">
      <section className="login-intro">
        <section
          className="black-layer d-flex flex-column justify-content-center align-items-center vh-100"
        >
          <h2 className="text-center font-weight-bold"> Welcome to
          <span className="d-inline-block px-2 my-2 ml-md-3">Reverse</span>
          </h2>
          <h3 className="text-center mt-1 mb-5">Sell your items now!</h3>
        </section>
      </section>

      <aside className="aside align-items-center align-self-center px-4 py-5">
        <form onSubmit={handleLogin} className="d-flex flex-column justify-content-center px-md-2">
          <h2 className="text-center my-5">Sign In</h2>

          <section className="mb-3">
            <AuthInput name="email" label="Email" type="email"
              onChange={(e) => setEmail(e.target.value)} value={email} required />
          </section>
          <section className="mb-3">
            <AuthInput name="password" label="Password" type="password"
              onChange={(e) => setPassword(e.target.value)} value={password} required />
          </section>

          <AuthButton text="Submit" />

          <Link to="/register"
            className="register-btn  mt-4 align-self-end text-decoration-none"
          >Don't have an account?</Link>
        </form>
      </aside>
    </main >
  );
};

export default Login;