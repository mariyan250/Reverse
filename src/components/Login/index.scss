@import '../../index.scss';

.login {
  display: flex;
  flex-direction: column;

  .login-intro {
    background: url('/img/login.jpg');
    background-position: 10%;
    background-size: cover;

    .black-layer {
      background: $background-black-transparent;
      height: 100%;
      display: flex;
      flex-direction: column;
      height: 100vh;
      justify-content: center;
      align-items: center;

      h2,
      h3 {
        color: $text-white;
        text-align: center;
        font-weight: 500;
      }

      h2 {
        font-size: 48px;
      }

      h3 {
        font-size: 28px;
        margin-top: $margin-xxs;
        margin-bottom: $margin-xxl;

      }

      span {
        background: $background-red;
        display: inline-block;
        padding: 0 $padding-xs;
        margin: $margin-xs 0;
      }
    }
  }

  .aside {
    padding: $padding-xxl $padding-l;

    form {
      display: flex;
      flex-direction: column;
      justify-content: center;
      padding: $padding-xxl 0;

      h2 {
        text-align: center;
        font-size: 42px;
        margin-top: $margin-xl;
        margin-bottom: $margin-xxl;
      }

      .auth-button {
        @include authButton;
      }

      .register-btn {
        margin: $margin-l 0 $margin-s;
        align-self: flex-end;
        font-size: 16px;
        text-decoration: none;
        color: $text-black;
      }
    }
  }
}

// Responsive
@media screen and (min-width: 500px) {
  .login {
    flex-direction: row;

    .login-intro {
      flex: 2.5;
      background-position: 0%;
    }

    .aside {
      flex: 1;
      margin: auto 0;
      padding: $padding-xxl;
    }
  }
}