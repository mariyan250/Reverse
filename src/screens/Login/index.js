// React and style
import React, { useState } from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import Input from 'shared/components/auth/Input';
import Button from 'shared/components/auth/Button';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const { t } = useTranslation();

  const handleLogin = (e) => {
    e.preventDefault();
    alert('Login!');
  };

  const inputs = [
    {
      name: 'email',
      label: t('login_screen.inputs.labels.email'),
      type: 'email',
      value: email,
      required: true,
      onChange(e) {
        setEmail(e.target.value);
      },
    },
    {
      name: 'password',
      label: t('login_screen.inputs.labels.password'),
      type: 'password',
      value: password,
      required: true,
      onChange(e) {
        setPassword(e.target.value);
      },
    },
  ];

  return (
    <main className="login d-flex vh-100 align-items-center justify-content-center px-4">
      <form onSubmit={handleLogin} className="d-flex flex-column">
        <h1 className="text-center mb-5">{t('login_screen.heading')}</h1>
        {inputs.map((input) => (
          <section className="mb-4">
            <Input {...input} />
          </section>
        ))}
        <Button text={t('login_screen.button')} />
        <Link
          to="/register"
          className="login-btn app-text-secondary mt-4 align-self-end text-decoration-none"
        >
          {t('login_screen.option_button')}
        </Link>
      </form>
    </main>
  );
};

export default Login;
