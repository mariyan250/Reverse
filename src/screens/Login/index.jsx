// React and style
import React, { useState } from 'react';
import styles from './index.module.scss';

// Translation
import { useTranslation } from 'react-i18next';

// Router
import { Link } from 'react-router-dom';

// Routes
import { routes } from 'shared/constants/routes';

// Components
import Input from 'shared/components/auth/Input';
import Button from 'shared/components/auth/Button';

function Login() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const { t } = useTranslation();

  const handleLogin = (e) => {
    e.preventDefault();
  };

  const inputs = [
    {
      name: 'email',
      label: t('login_screen.inputs.labels.email'),
      type: 'email',
      value: email,
      required: true,
      onChange: (e) => setEmail(e.target.value),
      testId: 'emailInput',
    },
    {
      name: 'password',
      label: t('login_screen.inputs.labels.password'),
      type: 'password',
      value: password,
      required: true,
      onChange: (e) => setPassword(e.target.value),
      testId: 'passwordInput',
    },
  ];

  return (
    <main className={`d-flex vh-100 align-items-center justify-content-center px-4`}>
      <form onSubmit={handleLogin} className={styles.form} data-testid="form">
        <h1 className={`text-center mb-5`}>{t('login_screen.heading')}</h1>

        {inputs.map((input, i) => (
          <div className={`mb-4`} key={i}>
            <Input {...input} />
          </div>
        ))}

        <div className={`mt-5`}>
          <Button text={t('login_screen.button')} />
        </div>

        <Link to={routes.register} className={`app-text-secondary d-inline-block mt-4 mb-3`}>
          {t('login_screen.option_button')}
        </Link>
      </form>
    </main>
  );
}

export default Login;
