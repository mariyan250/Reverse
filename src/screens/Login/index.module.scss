@import '../../index';

.form {
  max-width: to-rem(350px);
  margin-top: to-rem(120px);

  a {
    font-size: to-rem(16px);
  }
}
