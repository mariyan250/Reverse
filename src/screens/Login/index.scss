@import '../../index';

.login {
  .aside {
    form {
      h2 {
        font-size: to-rem(42px);
      }

      .register-btn {
        font-size: 16px;
        color: $text-black;
      }
    }
  }
}

@include media-breakpoint-md-up {
  .login {
    .login-intro {
      flex: 2.5;
      background-position: 0%;
    }

    .aside {
      flex: 1;
    }
  }
}
