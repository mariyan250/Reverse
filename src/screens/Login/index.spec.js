import React from 'react';
import { withWrapper } from 'tests/test-utils';
import { create } from 'react-test-renderer';
import Login from './index';

describe('Login', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const component = create(withWrapper(<Login />, initialState));
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
