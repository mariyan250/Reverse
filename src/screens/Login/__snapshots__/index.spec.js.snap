// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Rendering should render the component with initial state 1`] = `
<main
  className="d-flex vh-100 align-items-center justify-content-center px-4"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <h1
      className="text-center mb-5"
    >
      login_screen.heading
    </h1>
    <section
      className="mb-4"
    >
      <section
        className="input"
      >
        <label
          htmlFor="email"
        >
          login_screen.inputs.labels.email
        </label>
        <input
          className="p-1"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
        />
      </section>
    </section>
    <section
      className="mb-4"
    >
      <section
        className="input"
      >
        <label
          htmlFor="password"
        >
          login_screen.inputs.labels.password
        </label>
        <input
          className="p-1"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </section>
    </section>
    <section
      className="mt-5"
    >
      <button
        className="button app-bg-primary border-0 app-text-white w-100 p-2"
      >
        login_screen.button
      </button>
    </section>
    <a
      className="app-text-secondary d-inline-block mt-4 mb-3"
      href="/register"
      onClick={[Function]}
    >
      login_screen.option_button
    </a>
  </form>
</main>
`;
